Lib/ldap/dn.py

```diff
@@ -120,3 +120,33 @@ def is_dn(s,flags=0):
     return False
   else:
     return True
+
+
+def compare_dn(a, b):
+    """
+    Compares two distinguished names and return True if they are equal.
+    """
+    case_insensitive_attributes = [attr.lower() for attr in compare_dn.case_insensitive_attributes]
+
+    def normalize(attr, value):
+        if attr.lower() in case_insensitive_attributes:
+            value = value.lower()
+        return value
+    return [sorted((x.lower(), normalize(x, y), z) for x, y, z in rdn) for rdn in ldap.dn.str2dn(a)] == [sorted((x.lower(), normalize(x, y), z) for x, y, z in rdn) for rdn in ldap.dn.str2dn(b)]
+
+
+compare_dn.case_insensitive_attributes = []
```

Comment:
This global cache assumes that an application only contacts one LDAP server or only LDAP servers with compatible schema. The cache is invalid in case of incompatible schema.